From f98f5d4adc1c4bd4033638f6167c1bb95d642f89 Mon Sep 17 00:00:00 2001
From: Randall Hand <randall.hand@gmail.com>
Date: Thu, 23 Feb 2017 14:26:49 -0500
Subject: [PATCH] BugFix - data length

Even tho this is a "LONG" type, it's actually only 4 bytes. Typecasting
it to a long* caused a dereference beyond the end of the allocated
region.

issues#28
---
 lib/ytnef.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ytnef.c b/lib/ytnef.c
index c11c77c..afb2efb 100644
--- a/lib/ytnef.c
+++ b/lib/ytnef.c
@@ -1390,7 +1390,7 @@ void MAPIPrint(MAPIProps *p) {
           printf("] (%llu)\n", ddword_tmp);
           break;
         case PT_LONG:
-          printf("    Value: %li\n", *((long*)mapidata->data));
+          printf("    Value: %i\n", *((int*)mapidata->data));
           break;
         case PT_I2:
           printf("    Value: %hi\n", *((short int*)mapidata->data));
